src/PoolManager.sol

```diff
@@ -185,16 +185,20 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         IPoolManager.ModifyPositionParams memory params,
         bytes calldata hookData
     ) external override noDelegateCall onlyByLocker returns (BalanceDelta delta) {
+        PoolId id = key.toId();
+        if (pools[id].isNotInitialized()) revert PoolNotInitialized();
+
         if (key.hooks.shouldCallBeforeModifyPosition()) {
-            if (
-                key.hooks.beforeModifyPosition(msg.sender, key, params, hookData)
-                    != IHooks.beforeModifyPosition.selector
-            ) {
-                revert Hooks.InvalidHookResponse();
+            bytes4 selector = key.hooks.beforeModifyPosition(msg.sender, key, params, hookData);
```

Comment:
the length of the per-hook block expands once again https://github.com/Uniswap/v4-core/pull/404/files#r1397713947

I might tinker with some refactors here later

Reply:
Makes sense, I think both this and AccessLock introduce some spaghetti paths that we can clean up in a follow up?

Reply:
yeah I think we should wait to get both this and AccessLock in given they both change it - and then we can do that after :)